src/cargotype.cpp

```diff
@@ -198,11 +198,13 @@ void InitializeSortedCargoSpecs()
 
 	_standard_cargo_mask = 0;
 
-	_sorted_standard_cargo_specs_size = 0;
+	uint8 sorted_standard_cargo_specs_size = 0;
 	for (const auto &cargo : _sorted_cargo_specs) {
 		if (cargo->classes & CC_SPECIAL) break;
-		_sorted_standard_cargo_specs_size++;
+		sorted_standard_cargo_specs_size++;
 		SetBit(_standard_cargo_mask, cargo->Index());
 	}
+
+	_sorted_standard_cargo_specs = { _sorted_cargo_specs.data(), sorted_standard_cargo_specs_size };
```

Comment:
This is a vector, so you should just be able to convert that to a `span<>`? Not really sure why the size needed to be calculated first?

Reply:
Because we want a subset of the vector (most of the time it will be full vector though)

Reply:
Ah, the `break` there .. right .. that is sneaky :D

I think this deserves a comment somewhere :P